client/src/components/map.tsx

```diff
@@ -44,50 +53,100 @@ const mapConfig = {
     },
     {
       'id': 'blocks',
-      'type': 'line',
+      'type': 'fill',
       'source': 'custom',
-      'source-layer': 'BlockGroup',
+      'source-layer': 'blocks',
       'minzoom': 0,
       'layout': {
         'line-cap': 'round',
         'line-join': 'round',
       },
-
+      // 01=AL, 30=MT, 34=NJ, 35=NM, 36=NY
+      'filter': ['in', 'STATEFP10', '01', '30', '34', '35', '36'],
       'paint': {
-        'line-opacity': 0.6,
-        'line-color': 'red',
-        'line-width': 1,
+        'fill-color': [
+          'interpolate',
+          ['linear'],
+          ['to-number', [
+            'get',
+            'Score C (percentile)',
+          ]],
+          0.0,
+          'white',
+          1,
+          'blue',
+        ],
+        'fill-opacity': 0.75,
       },
     },
+    {
+      'id': 'labels-only',
+      'type': 'raster',
+      'source': 'labels',
+      'minzoom': 0,
+      'maxzoom': 22,
+    },
   ],
 };
 
+interface IMapWrapperProps {
+  features: Feature<Geometry>[],
+};
 
 // The below adapted from
 // https://taylor.callsen.me/using-openlayers-with-react-functional-components/
 const MapWrapper = ({features}: IMapWrapperProps) => {
   const [map, setMap] = useState<Map>();
   const [featuresLayer, setFeaturesLayer] = useState<VectorLayer>();
+  const [currentOverlay, setCurrentOverlay] = useState<Overlay>();
+  const [selectedFeature, setSelectedFeature] = useState<FeatureLike>();
 
   const mapElement = useRef() as
-        React.MutableRefObject<HTMLInputElement>;
+    React.MutableRefObject<HTMLInputElement>;
+  const mapRef = useRef() as React.MutableRefObject<Map>;
+
+
+  const popupContainer = React.useRef<HTMLDivElement>(null);
+  const popupCloser = React.useRef<HTMLAnchorElement>(null);
+  const popupContent = React.useRef<HTMLDivElement>(null);
+
+  const overlayRef = useRef() as React.MutableRefObject<Overlay>;
+  mapRef.current = map!;
```

Comment:
I'm not familiar with `map!` in this context. What is it doing?

Reply:
this is the non-null assertion operator ([ref](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator)) - you occasionally have to use this in Swift and I was surprised to find it in Typescript also - basically it's assuring the compiler that you know better and that this thing actually does exist.

That said, while this particular usage had the effect of temporarily silencing the compiler, in this case it actually might not exist, so this should be handled differently. In general we should move away from using `ref`s in this way - it works but there are better ways to do this.

Reply:
I can look into other solutions and push a spot fix for this

Reply:
I've never seen it used to stop linter/compiler complaints this way. I'm fine with it with maybe a comment or `todo`.